src/components/Api.js

```diff
@@ -0,0 +1,84 @@
+export default class Api {
+  constructor(baseUrl, header) {
+    this._baseUrl = baseUrl;
+    this._header = header;
+  }
+
+  _checkPromise(res) {
```

Comment:
It's great to have a method for processing the server response, so the logic is not duplicated